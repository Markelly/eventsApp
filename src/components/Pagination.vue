<template>
  <Paginate
    :page-count="totalPages"
    :click-handler="gotoPage"
    prev-text="Previous"
    next-text="Next"
    container-class="pagination"
    page-class="pagination__item"
    prev-class="pagination__item"
    next-class="pagination__item"
    data-pagination
  />
</template>

<script>
import VuejsPaginate from "vuejs-paginate";
export default {
  name: "Pagination",
  components: {
    Paginate: VuejsPaginate
  },
  props: {
    totalPages: {
      type: Number,
      default: 0
    }
  },
  methods: {
    gotoPage(page) {
      this.$emit("gotoPage", page);
    }
  }
};
</script>

<style lang="scss">
@import "../styles/events-app";

.pagination {
  list-style-type: none;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  color: $color6;
  padding: 0;
  font-weight: 600;
  margin: 48px 12px;

  &__item {
    border-radius: 6px;
    padding: 8px 16px;
  }

  .active,
  &__item:hover {
    background-color: $color6;
    color: $color2;
  }

  .disabled,
  .disabled:hover {
    color: $color3;
    background: $color1;
    cursor: not-allowed;
  }
}

.active,
.pagination__item:hover {
  background-color: $color6;
  color: $color2;
}
</style>
