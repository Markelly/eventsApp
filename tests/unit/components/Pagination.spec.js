import Pagination from "../../../src/components/Pagination";
import { createLocalVue, shallowMount } from "@vue/test-utils";

describe("<Pagination/>", () => {
  const localVue = createLocalVue();
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Pagination, {
      localVue,
      propsData: {
        totalPages: 10
      }
    });
  });

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should emit gotoPage event", () => {
    wrapper.vm.gotoPage(3);
    expect(wrapper.emitted("gotoPage", 3)).toBeTruthy();
  });
});
