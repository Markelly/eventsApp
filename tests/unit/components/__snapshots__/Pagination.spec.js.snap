// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> should render the component 1`] = `<paginate-stub pagecount="10" clickhandler="function () { [native code] }" pagerange="3" marginpages="1" prevtext="Previous" nexttext="Next" breakviewtext="â€¦" containerclass="pagination" pageclass="pagination__item" prevclass="pagination__item" nextclass="pagination__item" activeclass="active" disabledclass="disabled" firstbuttontext="First" lastbuttontext="Last" data-pagination=""></paginate-stub>`;
